refactor: build script & remove relative path imports

ananas-block opened the pull request:
- replaced relative path imports for npm packages
- added the use of yarn link to build script
- bumped package versions
- regenerated yarn.lock files

SwenSchaeferjohann commented on scripts/build.sh:
```diff
@@ -45,6 +52,12 @@ cleanup_and_install() {
+  
+
+  
+  if [ "${dir}" != "light-system-programs" ]; then
+      yarn link
+  fi
```
why do we `yarn link` cli, light-circuits, and relayer?

vadorovsky replied:
`yarn link` without arguments consumes the links in the current project. Doing `yarn link` in cli, applies the @lightprotocol/zk.js (and other links we made) in the cli project. Without it, cli would still try to use the release specified in package.json, so it would end up downloading it from npmjs.

SwenSchaeferjohann replied:
afaik that's incorrect. `yarn link` without arguments makes your package available to all other local projects to consume. the script uses `yarn link light-zk.ks` inside the consuming repos

SwenSchaeferjohann replied:
the script ordering may be confusing since yarn link happens at the end; the reason it works is that we're executing the scripts first with light-prover, then zk.js, and lastly the consuming projects

ananas-block replied:
The syntax yarn link @lightprotocol/zk.js and @lightprotocol/prover.js works for me.

SwenSchaeferjohann replied:
it's still not clear to me:
> why do we `yarn link` cli, light-circuits, and relayer?

ananas-block replied:
it's not necessary but it also doesn't do anything

vadorovsky replied:
> afaik that's incorrect. `yarn link` without arguments makes your package available to all other local projects to consume. the script uses `yarn link light-zk.ks` inside the consuming repos

OK, you are right. Sorry!

SwenSchaeferjohann commented:
- why do we have an anchor flag if the flag is always set to `false`?

- I think there may be a potential foot gun with how we use the CLI to start the test-validator and execute tests in relayer dir; it seems we don't actually use the locally linked version of the CLI, but a globally installed one

vadorovsky reviewed (changes requested):
Agree with other @SwenSchaeferjohann's comments. I think we can just get rid of the `--yarn` and `--anchor` flags in `build.sh`. It seems like yarn ultimately won as the build system for this project and we can just `yarn build` everything. :stuck_out_tongue:

ananas-block commented:
- removed the anchor flag
- light-circuits should not be built by default thus I added an exception

> * it seems we don't actually use the locally linked version of the CLI, but a globally installed one
@SwenSchaeferjohann why do you get that impression?
We don't want to use a global installed cli but the cli built in light-protocol/cli.

SwenSchaeferjohann commented:
> * removed the anchor flag
> * light-circuits should not be built by default thus I added an exception
> 
> > * it seems we don't actually use the locally linked version of the CLI, but a globally installed one
> 
> @SwenSchaeferjohann why do you get that impression? We don't want to use a global installed cli but the cli built in light-protocol/cli.

nvm, i was erring; we don't specify cli as a dependency however we do run the cli commands based on local  path

SwenSchaeferjohann commented on scripts/build.sh:
```diff
@@ -36,22 +28,29 @@ cleanup_and_install() {
 
-  if [ "${anchor}" = true ] ; then
-    light-anchor build
+  if [ "${dir}" != "light-system-programs" ]; then
+      yarn link
   fi
```
```suggestion
yarn link
```

SwenSchaeferjohann replied:
fine, then this would be less confusing
